ibis/expr/types/temporal.py

```diff
@@ -474,6 +474,43 @@ def delta(
         """
         return ops.DateDelta(left=self, right=other, part=part).to_expr()
 
+    def epoch_days(self) -> ir.IntegerValue:
+        """Return the number of days since the UNIX epoch date.
+
+        Examples
+        --------
+        >>> from datetime import date
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> d = date(2020, 1, 1)
+        >>> ibis.date(d)
+        ┌────────────┐
+        │ 2020-01-01 │
+        └────────────┘
+        >>> ibis.date(d).epoch_days()
+        ┌───────┐
+        │ 18262 │
+        └───────┘
+        >>> t = ibis.memtable({"date_col": [d]})
```

Comment:
Slightly simpler example, no need to involve `datetime.date` in here at all.

```suggestion
        >>> import ibis
        >>> ibis.options.interactive = True
        >>> date = ibis.date(2020, 1, 1)
        >>> date
        ┌────────────┐
        │ 2020-01-01 │
        └────────────┘
        >>> date.epoch_days()
        ┌───────┐
        │ 18262 │
        └───────┘
        >>> t = date.name("date_col").as_table()
```